pkg/aws/ec2.go

```diff
@@ -231,6 +241,18 @@ func AssertEC2TagValue(t *testing.T, ctx context.Context, client EC2Client, inpu
 	assert.True(t, hasMatch, "Tag with key '%s' does not exist.", input.TagName)
 }
 
+// AssertEC2VolumeTagValue asserts that an EBS Volumes has a tag with the given value.
+func AssertEC2VolumeTagValue(t *testing.T, ctx context.Context, client EC2Client, input AssertEC2VolumeTagValueInput) {
+	volume, _ := getEC2VolumeByVolumeIDE(ctx, client, input.VolumeId)
+	for _, tag := range volume.Tags {
+		tagKey := *tag.Key
+		tagValue := *tag.Value
+		if tagKey == input.TagName {
+			assert.Equal(t, input.Value, tagValue, "Tag with key '%s' does not match expected value.", tagKey)
+		}
+	}
+}
```

Comment:
**Suggestion:** If the volume does not have the specified tag at all (regardless of the value), the test should fail. (Unit tests should cover this scenario as well.)

Reply:
added .